cwms-data-api/src/main/java/cwms/cda/data/dto/location/kind/Embankment.java

```diff
@@ -106,24 +102,10 @@ public LookupType getDownstreamProtectionType() {
         return downstreamProtectionType;
     }
 
-    public Location getLocation() {
-        return location;
-    }
-
-    public LocationIdentifier getProjectId() {
-        return projectId;
-    }
-
     @Override
     public void validate() throws FieldException {
-        if (this.location == null) {
-            throw new FieldException("Location field can't be null");
-        }
-        location.validate();
-        if (this.projectId == null) {
-            throw new FieldException("Project location Id field can't be null");
-        }
-        projectId.validate();
+        super.validate();
```

Comment:
I can't modify the validate API, since that affects 66 impls.  What I did was create a method that returns a list of missing values.  It catches the exceptions from the CwmsId and Location and collects their missing fields from the exceptions they throw.